lib/strings.nix

```diff
@@ -356,7 +356,7 @@ rec {
   */
   toShellVar = name: value:
     lib.throwIfNot (isValidPosixName name) "toShellVar: ${name} is not a valid shell variable name" (
-    if isAttrs value then
+    if isAttrs value && ! isCoercibleToString value then
```

Comment:
Needs doc update, like

```diff
-     Supported value types are strings (mapped to regular variables), lists of strings
+     Supported value types are strings (mapped to regular variables), string-coercible attrsets
+     (also mapped to regular variables), lists of strings
```

Reply:
Did you change it? The current text is too implicit.

It's also better to use singular where possible, as it avoids ambiguity.

Reply:
I think it's just the right amount of implicit: the intent is that strings are mapped to regular variables, and, as a side-note, things that should obviously be treated as strings are treated as strings. Mentioning string-coercible attrsets feels heavy and impedes readability IMO.

:+1: for singular

Reply:
Fair enough.